Lib/test/test_tarfile.py

```diff
@@ -690,6 +691,36 @@ def format_mtime(mtime):
             tar.close()
             os_helper.rmtree(DIR)
 
+    @unittest.skipUnless(hasattr(errno, "EFTYPE"), "errno.EFTYPE required")
+    def test_extract_chmod_eftype(self):
+        # Extracting a file as non-root should skip the sticky bit (gh-108948)
+        # even on platforms where chmod fails with EFTYPE (i.e. FreeBSD). But
+        # we need to take care that any other error is preserved.
+        mode = "-rwxrwxrwt"
+        with ArchiveMaker() as arc:
+            arc.add("sticky1", mode=mode)
+            arc.add("sticky2", mode=mode)
+        tar = arc.open(errorlevel=2)
+        DIR = os.path.join(TEMPDIR, "chmod")
+        os.mkdir(DIR)
+        self.addCleanup(os_helper.rmtree, DIR)
+        with tar:
+            # this should not raise:
+            tar.extract("sticky1", DIR, filter="fully_trusted")
```

Comment:
I think that you should split this test in two tests:

* a method which doesn't mock anything and jsut check that tarfile works as expected: this line.
* another method which mocks chmod() to test the complicated EFTYPE code path.